File: backend/node-express/src/repositories/user.repository.js
const { prisma } = require('../config/database');

class UserRepository {
  async findByUsername(username) {
    return prisma.user.findUnique({ where: { username } });
  }

  async findById(id) {
    return prisma.user.findUnique({ where: { id } });
  }

  async create(data) {
    return prisma.user.create({ data });
  }
}

module.exports = new UserRepository();

File: backend/dot-net/Repositories/UserRepository.cs
using Microsoft.EntityFrameworkCore;
using backend.Data;
using backend.Models;

namespace backend.Repositories;

public class UserRepository
{
    private readonly AppDbContext _context;

    public UserRepository(AppDbContext context)
    {
        _context = context;
    }

    public User? FindByUsername(string username)
    {
        return _context.Users.FirstOrDefault(u => u.Username == username);
    }

    public User? FindById(string id)
    {
        return _context.Users.FirstOrDefault(u => u.Id == id);
    }

    public User Create(string username, string password)
    {
        var user = new User
        {
            Username = username,
            Password = password
        };
        _context.Users.Add(user);
        _context.SaveChanges();
        return user;
    }
}
